pkg/controllers/workloadrebalancer/workloadrebalancer_controller.go

```diff
@@ -110,6 +147,8 @@ func (c *RebalancerController) doWorkloadRebalance(ctx context.Context, rebalanc
 	newStatus = rebalancer.Status.DeepCopy()
```

Comment:
Seems this deep copy is redundant as it will be overwritten by the following `buildWorkloadRebalancerStatus` or `syncWorkloadsFromSpecToStatus`.